### DISCUSSION TITLE
How to upgrade the CGNS version or Use CGNS that I installed myself

### DISCUSSION THREAD

**Discussion by Zcaic** [su2code/SU2#2064] [Q&A]
Hello,
I have a cgns file but it‘s version seems to be incompatible with the CGNS of the SU2.
How to upgrade the CGNS version or Use CGNS that I installed myself when I build SU2?

**Comment by pcarruscag** [comment]
See externals/cgns/meson.build line 70, that is where we define the cgns dependency, you need to consult the meson documentation for how to do something similar pointing to your version.
For a similar example see SU2/meson.build line 174.

**Reply by Zcaic** [reply]
Thank you very much for replying to me！
I'll give it a try as you say.

**Reply by Zcaic** [reply]
@pcarruscag 
Thank you for the guidance. I've solved this issue. I change externals/cgns/meson.build line 70 "**cgns_dep = declare_dependency(link_with: lib_cgns,dependencies: hdf5_dep,include_directories: cgns_include)**" to "**cgns_dep = declare_dependency(link_args:['-L/usr/lib','-l:libcgns.so.4.3','-l:libhdf5.so.310'],include_directories:['/usr/include'])**"

Then it compiled successfully and It converts CGNS static library linking into dynamic library linking.
